modules/tensor_mechanics/src/materials/RadialReturnStressUpdate.C

```diff
@@ -103,6 +106,31 @@ RadialReturnStressUpdate::updateState(RankTwoTensor & strain_increment,
    * currently just the elasticity tensor, irrespective of compute_full_tangent_operator
    */
   tangent_operator = elasticity_tensor;
+
+  if (compute_full_tangent_operator && scalar_effective_inelastic_strain != 0.0)
+  {
+    // mu = _three_shear_modulus / 3.0;
+    // effective_trial_stress = von mises stress = std::sqrt(3.0 / 2.0) * norm_dev_stress
+    // scalar_effective_inelastic_strain = Delta sigma^cr_n+1
+    // _dstrain_dstress = derivative of scalar_effective_inelastic_strain w.r.t. von mises stress
+
+    const RankTwoTensor deviatoric_stress = stress_new.deviatoric();
+    const Real norm_dev_stress =
+        std::sqrt(deviatoric_stress.doubleContraction(deviatoric_stress)); // ||s_n+1||
+    const RankTwoTensor flow_direction = deviatoric_stress / norm_dev_stress;
+    const RankFourTensor flow_direction_cross = flow_direction.outerProduct(flow_direction);
+    const Real deriv =
+        -(computeDerivative(effective_trial_stress, scalar_effective_inelastic_strain) + 1.0) /
+        _three_shear_modulus;
+
+    const RankFourTensor A =
+        _ifoursym + _three_shear_modulus *
+                        (scalar_effective_inelastic_strain / std::sqrt(1.5) / norm_dev_stress *
+                             (_ifoursym - flow_direction_cross) * _ifourdev +
+                         deriv * flow_direction_cross);
```

Comment:
Wow - those are some dense operations!  I see lots of division operators in here. You'll have to excuse my ignorance, but are there any chances for division by zero under _any_ circumstance? If so, you'll probably want to add a few `mooseAsserts()` in here to make sure our assumptions are met when using this code.

Reply:
Yeah, that's a good point. I think the only one that matters is making sure `norm_dev_stress` isn't zero. I can add a check for that, which would effectively skip everything else.

Reply:
`mooseAssert(std::sqrt(1.5) != 0.0, "Math is broken!");`

Reply:
Divide by zero is protected in #11241. I'll rebase when that is merged.

Reply:
Protected `norm_dev_stress` via #11241 . Added assert for `_three_shear_modulus`, although it is somewhat unnecessary since there wouldn't be creep at all if the shear modulus was zero.

Reply:
I think I'd still put in an assert on `norm_dev_stress` just to make sure it's really protected. That doesn't cost us anything in the opt executable.